### ISSUE TITLE
Segfault when using reoptimization and concurrent solving

### ISSUE THREAD

**Issue by waweber** [scipopt/PySCIPOpt#884]
**Describe the bug**
A segfault occurs during reoptimization when using concurrent solving.

**To Reproduce**
Run this program:
```python
from pyscipopt import Model

model = Model()
x = model.addVar(lb=0.0, ub=10.0)
y = model.addVar(lb=0.0, ub=10.0)
z = x + y
model.addCons(y >= 2*x)
model.setObjective(z, "maximize")

model.enableReoptimization()
model.solveConcurrent()

print(f"x = {model.getVal(x)}")
print(f"y = {model.getVal(y)}")
print(f"z = {model.getVal(z)}")

model.freeReoptSolve()

model.addCons(x <= 3)
model.solveConcurrent()

print(f"x = {model.getVal(x)}")
print(f"y = {model.getVal(y)}")
print(f"z = {model.getVal(z)}")
```

The issue occurs with any combination of solving methods that uses `solveConcurrent()`

**Expected behavior**
The program runs the same way as when normal `model.optimize()` is called.

**System**
 - Fedora Linux 40
 - SCIP from master branch
 - PySCIPOpt compiled from master branch

**Additional context**
Not 100% sure if this is from PySCIPOpt or a bug in SCIP itself

**Comment by Joao-Dionisio** [comment]
Hey @waweber! Apologies for the delay, August-time is always difficult.

Are there any updates on this? You can use solveConcurrent normally if you don't reoptimize, right?

**Comment by waweber** [comment]
Correct, I can use solveConcurrent without reoptimization, or reoptimize without solveConcurrent, but not both.

Notably, if I use solveConcurrent on the initial optimization, then use the single threaded optimize when reoptimizing, it also segfaults.

**Comment by Joao-Dionisio** [comment]
Hey @waweber ! Sorry for the nearly 2-year long delay!!!!!!!!!

I digged a bit into the SCIP code, and it turns out that there is a segmentation fault: reoptimization and concurrent solving are really not compatible. There is some cleanup shenanigans going on that subsequent solvers invalidly try to access. It would be pretty cool to be able to do both, sure, but it's seems to be shaping up to be a quite considerable rework. As such, we're thinking of just retunring an error, for now.

**Comment by Joao-Dionisio** [comment]
The error is now live [here](https://github.com/scipopt/scip/compare/d99a0d20351d224ce2a9f119f90fd135ef9be17c...265afb29c76d6753d01a7342e7d47c83ded4e6a7). Maybe one day we come back to this and work on proper integration of the two features, but for now this indeed seemed like the best option.
